### PR TITLE
Add ESmini simulator

### PR DESCRIPTION
**Pull request by cassava**
Still needs to be rebased, but otherwise ready for review.

### CODE REVIEW

**Review by cassava — COMMENTED** [review]
Since I am placing this PR, I cannot approve. But I did review all the code from Tobias and fixed several findings I had. In all, looks good!

File: osi/include/cloe/component/osi_sensor.hpp
[the remark this answers is not in the document]
Comment (cassava) [reply osi/include/cloe/component/osi_sensor.hpp:1]: On `develop`, we have the following definition:

```cpp
virtual const osi3::SensorData& get() const = 0;
```

This differs in that a reference is returned, compared to a `shared_ptr<osi3::SensorData>`.

[the remark this answers is not in the document]
Comment (cassava) [reply osi/include/cloe/component/osi_sensor.hpp:1]: It actually looks like the `OsiSensor` is only used by a test right now. That can't be right...

**Review by hvolx — COMMENTED** [review]
osi_transceiver_tcp needs to be fixed but for for EsMini it does not matter currently --> osi_transceiver_tcp may be fixed in the future!

File: osi/include/cloe/component/osi_sensor.hpp
[the remark this answers is not in the document]
Comment (cassava) [reply osi/include/cloe/component/osi_sensor.hpp:1]: Fixed.

File: osi/CMakeLists.txt
[the remark this answers is not in the document]
Comment (cassava) [reply osi/CMakeLists.txt:1]: Leaving as is for now.

**Pull request merged**